frontend/src/Map/CrossingMap.js

```diff
@@ -36,29 +49,39 @@ class CrossingMap extends React.Component {
       return (<div>Error Loading Crossings</div>);
     }
 
-    var openCrossings = this.props.data.allCrossings.nodes.filter((crossing) => crossing.statusUpdateByLatestStatusId.statusId === 1);
-    var closedCrossings = this.props.data.allCrossings.nodes.filter((crossing) => crossing.statusUpdateByLatestStatusId.statusId === 2);
+    const allCrossings = this.props.data.allCrossings.nodes;
 
     return (
       <Map
         onStyleLoad={this.onMapboxStyleLoad.bind(this)}
+        onClick={this.onMapClick.bind(this)}
         style={mapboxstyle}
         containerStyle={{
           height: this.props.mapHeight,
           width: this.props.mapWidth,
           display: "block"
         }}
-        fitBounds={this.props.viewport}>
-
+        fitBounds={this.props.viewport}
+        center={this.state.center}>
         <Layer
           type="symbol"
           id="closedCrossings"
           layout={{ 'icon-image': 'cross-15', 'icon-allow-overlap': true }}
+          layerOptions={{"filter":
+            [
+              "all",
+              ["!=", "id", this.state.selectedCrossingId],
+              ["==", "crossingStatus", 2]
```

Comment:
I don't understand what these layerOptions do and it wasn't very specific in the docs.
https://github.com/alex3165/react-mapbox-gl/blob/master/docs/API.md

I see we use a fixed number here, 2 and 1 somewhere else. What are these magic numbers?

Reply:
The LayerOptions object gets passed through the react-mapbox-gl wrapper into the mapbox map. I'm passing in the [filter](https://www.mapbox.com/mapbox-gl-js/style-spec/#types-filter).